app/service/planning_svc.py

```diff
@@ -172,8 +172,10 @@ async def get_links(self, operation, buckets=None, agent=None, trim=True):
         if agent:
             links.extend(await self.generate_and_trim_links(agent, operation, abilities, trim))
         else:
+            agent_links = []
             for agent in operation.agents:
-                links.extend(await self.generate_and_trim_links(agent, operation, abilities, trim))
+                agent_links.append(await self.generate_and_trim_links(agent, operation, abilities, trim))
+            links = self._cross_check_agents_for_duplicate_singletons(agent_links)
```

Comment:
wouldn't we want to extend links here instead of overwrite?

Reply:
Extending links would just create a single master list of links, but there could be duplicates of the same restricted ability across various agents (leading to duplicate lateral movement and agents). Keeping everything separate for each agent makes it a lot easier to deduplicate based on command while flattening it manually down to the pruned master list, which is what the `_cross_check...` method is doing. In theory, you might be able to do this in place with a lambda or comparator, but I'm pretty sure it would be unhelpfully long and complex.

Reply:
ah thanks I see that now